server/storage/database.go

```diff
@@ -78,3 +87,28 @@ func (db *MySQLStorage) Delete(gun string) error {
 	_, err := db.Exec(stmt, gun)
 	return err
 }
+func (db *MySQLStorage) GetTimestampKey(gun string) (cipher string, public []byte, err error) {
+	stmt := "SELECT `cipher`, `public` FROM `timestamp_keys` WHERE `gun`=?;"
+	row := db.QueryRow(stmt, gun)
+
+	err = row.Scan(&cipher, &public)
+	if err == sql.ErrNoRows {
+		return "", nil, ErrNoKey{gun: gun}
+	} else if err != nil {
+		return "", nil, err
+	}
+
+	return cipher, public, err
+}
+func (db *MySQLStorage) SetTimestampKey(gun, cipher string, public []byte) error {
+	stmt := "INSERT INTO `timestamp_keys` (`gun`, `cipher`, `public`) VALUES (?,?,?);"
+	_, err := db.Exec(stmt, gun, cipher, public)
+	if err, ok := err.(*mysql.MySQLError); ok {
+		if err.Number == 1022 { // duplicate key error
```

Comment:
Unfortunately not. I had to go dig this error out of the MySQL docs. The library takes the raw output from MySQL and parses it into its error object which includes the `Number` field.